imap_processing/cli.py

```diff
@@ -281,10 +415,18 @@ def main():
     #       this function for an entrypoint.
     args = _parse_args()
 
-    _validate_args(args)
+    print(args)
 
+    _validate_args(args)
     cls = getattr(sys.modules[__name__], args.instrument.capitalize())
-    instrument = cls(args.level, args.file_path)
+    instrument = cls(
+        args.data_level,
+        args.dependency,
+        args.start_date,
+        args.end_date,
+        args.version,
+        args.upload_to_sdc,
+    )
```

Comment:
a dump question, how does say `--start-date` gets converted to `args.start_date`? I looked but I am not seeing it where this conversion happens.

Reply:
You know, I left it this way with the intent to go back to it, and then it all worked so I never did. Looks like `args` automatically converts the arguments from a hyphen to an underscore.